stdlib/experimental/geo/README.md

```diff
@@ -254,14 +292,231 @@ asTracks = (tables=<-, groupBy=["id","tid"], orderBy=["_time"]) =>
+- `token` - source token
+- `point` - source coordinates
+- `level` - requested cell level of the target token
 
 Either `token` or `point` must be specified.
 
 Example:
-```
+```js
 t = geo.s2CellIDToken(point: {lat: 40.51757813, lon: -73.65234375}, level: 10)
 ```
+
+### Function `s2CellLatLon`
+
+Returns coordinates of the S2 cell center.
+
+Input parameters are:
+- `token` - cell ID token
+
+Example:
+```js
+ll = geo.s2CellLatLon(token: "89c284")
+```
+
+### Function `ST_Contains`
+
+Returns boolean value whether the region contains geometry or not.
+Parameter `geometry` can be either a point or a linestring.
+
+Input parameters are:
+- `region`
+- `geometry`
+
+Example:
+```js
+box = {minLat: 40.51757813, maxLat: 40.86914063, minLon: -73.65234375, maxLon: -72.94921875}
+
+from(bucket:"mta")
+    ...
+    |> geo.toRows()
+    |> map(fn: (r) => ({
+      r with st_contains: ST_Contains(region: region, geometry: {lat: r.lat, lon: r.lon})
+    }))
+```
+
+```js
+box = {minLat: 40.51757813, maxLat: 40.86914063, minLon: -73.65234375, maxLon: -72.94921875}
+
+from(bucket:"mta")
+    ...
+    |> geo.toRows()
+    |> geo.asTracks()
+    |> geo.ST_LineString()
+    |> map(fn: (r) => ({
+      r with st_contains: ST_Contains(region: region, geometry: {linestring: r.st_linestring})
+    }))
+```
+
+### Function `ST_Distance`
+
+Returns distance between specified region and geometry.
+Parameter `geometry` can be either a point or a linestring.
+
+Input parameters are:
+- `region`
+- `geometry`
+
+Example:
+```js
+box = {minLat: 40.51757813, maxLat: 40.86914063, minLon: -73.65234375, maxLon: -72.94921875}
+
+from(bucket:"mta")
+    ...
+    |> geo.toRows()
+    |> map(fn: (r) => ({
+      r with st_distance: ST_Distance(region: region, geometry: {lat: r.lat, lon: r.lon})
+    }))
+```
+
+### Function `ST_DWithin`
+
+Returns boolean if geometry is within a distance to specified region.
+Parameter `geometry` can be either a point or a linestring.
+
+Input parameters are:
+- `region`
+- `geometry`
+
+Example:
+```js
+box = {minLat: 40.51757813, maxLat: 40.86914063, minLon: -73.65234375, maxLon: -72.94921875}
+
+from(bucket:"mta")
+    ...
+    |> geo.toRows()
+    |> map(fn: (r) => ({
+      r with st_within: ST_DWithin(region: region, geometry: {lat: r.lat, lon: r.lon}, distance: 15.0)
+    }))
+```
+
+#### Function definition
+
+```js
+ST_DWithin = (region, distance, geometry) =>
+  ST_Distance(region: region, geometry: geometry) <= distance
+```
+
+### Function `ST_Intersects`
+
+Returns boolean whether geometry intersects specified region.
+Parameter `geometry` can be either a point or a linestring.
+
+Example:
+```js
+box = {minLat: 40.51757813, maxLat: 40.86914063, minLon: -73.65234375, maxLon: -72.94921875}
+
+from(bucket:"mta")
+    ...
+    |> geo.toRows()
+    |> geo.asTracks()
+    |> geo.ST_LineString()
+    |> map(fn: (r) => ({
+      r with st_intersects: ST_Intersects(region: region, geometry: {linestring: r.st_linestring})
```

Comment:
I think you mean `region: box` here too.

Reply:
Absolutely. Thank you for such thorough review.